### PR TITLE
Use cases for Tasks Management

### PR DESCRIPTION
**Pull request by wesho1107**
Added following use cases:
- Add task feature
- View tasks list feature
- Update task progress feature

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -173,7 +173,7 @@ Below is a class diagram describing the implementation of `Session` and its resp
 
 **Alternative 1 (current choice):** The `CreateSession` feature takes in arguments of varying number of student names.
 - Pros:This choice offers a high degree of flexibility when it comes to adding students to a session. You can add any number of students when creating a session, which is essential for accommodating different class sizes and situations.
-**Alternative 2:** An alternative design could involve creating two separate constructor methods within the `CreateSession` class. One constructor would be responsible for adding a student to an existing `SessionStudents`, and the other would take an entire `SessionStudents` object as an argument.
+  **Alternative 2:** An alternative design could involve creating two separate constructor methods within the `CreateSession` class. One constructor would be responsible for adding a student to an existing `SessionStudents`, and the other would take an entire `SessionStudents` object as an argument.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:176]: this space might be accidental?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -289,15 +289,15 @@ GradedTest testFromObjects = new GradedTest(
...
-GradedTest testFromString = new GradedTest("RA1:90 | RA2:85 | MidTerms:75 | Finals:80 | PE:95");
+        GradedTest testFromString = new GradedTest("RA1:90 | RA2:85 | MidTerms:75 | Finals:80 | PE:95");
 
 // Check if both objects are equal
-assertEquals(testFromObjects, testFromString);
+        assertEquals(testFromObjects, testFromString);
 ```
 </box>
 
 - Cons:
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:299]: Sorry boss

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -894,9 +918,74 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 1c. Invalid fields provided
+
+  * 1c1. F.A.K.E.J.A.R.V.I.S. shows an error message.
+
+    Use case ends.
+
+* 2a. The list is empty.
+
+  Use case ends.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md:957]: add an indent here for `Use case ends.`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -894,9 +918,74 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+
+**Use case: Update task progress**
+
+**MSS**
+
+1.  User requests to update a task progress
+2.  F.A.K.E.J.A.R.V.I.S. updates task progress and display updated progress.
```
Comment (Respirayson) [inline docs/DeveloperGuide.md]: displays* but small grammar nit

**Review by Respirayson — APPROVED** [review]
LGTM

**Review by raydenlim — APPROVED** [review]
Apart from some small details, LGTM!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (wesho1107) [reply docs/DeveloperGuide.md:957]: This is fine is it not? It ends at 2a, not a deeper hierarchy

**Pull request merged**